### PR TITLE
Add basic secrets processing for GitHub integration

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

File: cloud/gcp/bin/clean [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -e
+
+echo Cleaning existing state...
+
+terraform state rm `terraform state list`
```
Comment (pisuke) [inline cloud/gcp/bin/clean:5]: The terraform state is kept in the remote backend bucket, it must not be erased

**Comment by grafnu** [comment]
This is just a utility function (clean) -- if you don't call it then it
won't be used/removed!   Maybe it's not useful... I still can't get my
system to work so it's not clear to me what *does* need to be cleaned out
to get to stable state.

File: cloud/gcp/bin/init [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+for file in main.tf udmi-sites.tf terraform.tfvars; do
+    if [[ ! -f $file ]]; then
+        echo Configuration file $file does not exist, did you run bin/setup?
+        false
+    fi
+done
+
+terraform init $*
+
+terraform import google_project.udmi-project ${GCP_PROJECT_NAME:?}
```
Comment (pisuke) [inline cloud/gcp/bin/init:16]: This must be removed because the state already includes the project

File: cloud/gcp/bin/init [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+terraform import google_storage_bucket.tf-bucket ${GCP_PROJECT_NAME}-terraform
```
Comment (pisuke) [inline cloud/gcp/bin/init:18]: This must be removed because the state already includes the storage bucket

**Comment by grafnu** [comment]
OK -- this was what you had in the readme!

**Comment by pisuke** [comment]
> This is just a utility function (clean) -- if you don't call it then it won't be used/removed! Maybe it's not useful... I still can't get my system to work so it's not clear to me what *does* need to be cleaned out to get to stable state.

It looks dangerous to me. If we lose the state, we must start from scratch, which is not what we want. I would remove it to avoid misuse.

**Comment by grafnu** [comment]
Ok -- I took it out for now, but at some state we'll need something that
cleans out the local state on a machine -- not sure what that would be,
exactly.  Maybe it's removing the ".terraform" directory?  Basically, it
needs to undo whatever might have been done with the local setup... (but
that can be added later)

**Pull request closed without merging**